src/components/table-hoc/reducers/TableWithSortReducers.ts

```diff
@@ -0,0 +1,54 @@
+import * as _ from 'underscore';
+import {IReduxAction} from '../../../utils/ReduxUtils';
+import {ITableHeaderAddPayload, ITableHeaderBasePayload, TableHeaderActionTypes} from '../actions/TableHeaderActions';
+
+export interface ITableWithSortState {
+    id: string;
+    tableId: string;
+    isAsc: boolean;
+}
+
+const addTableHeaderReducer = (state: ITableWithSortState[], action: IReduxAction<ITableHeaderAddPayload>) => {
+    return [
+        ...state,
+        {
+            id: action.payload.id,
+            tableId: action.payload.tableId,
+            isAsc: action.payload.isDefault === true ? true : undefined,
+        },
+    ];
+};
+
+const removeTableHeaderReducer = (state: ITableWithSortState[], action: IReduxAction<ITableHeaderBasePayload>) => {
+    return _.reject(state, (header: ITableWithSortState) => header.id === action.payload.id);
+};
+
+const sortTableHeaderReducer = (state: ITableWithSortState[], action: IReduxAction<ITableHeaderBasePayload>) => {
+    const current = _.findWhere(state, {id: action.payload.id});
+    if (current) {
+        return _.map(state, (header: ITableWithSortState) => {
+            if (header.id === current.id) {
+                return {
+                    ...header,
+                    isAsc: !header.isAsc,
+                };
+            }
+            return header.tableId === current.tableId ? {...header, isAsc: undefined} : header;
+        });
+    }
+    return state;
+};
+
+const TableWithSortActionReducers: {[key: string]: (...args: any[]) => any} = {

```

Comment:
we use 2 kind of structure for the reducer, I dont have preference but I dont think its a good idea to use both.  When we use the switch case and when this one ?

Reply:
Personally I really prefer this one. The problem with the switch case is that you can't reuse let and const name between cases and that the action is not correctly typed